src/alda/lisp/attributes.clj

```diff
@@ -57,10 +57,62 @@
               (pos? x))]}
   (constantly x))
 
+;; Converts string/number representation of note length into number of beats
+(defn- parse-note-length
+  [length]
+  {:pre [(or (string? length) (number? length))]}
+  (cond
+    (string? length)
+    (let [[_ number _ dots]  (re-matches #"(\d+(\.\d+)?)(\.*)" length)]
+      (if number
+        (:value (note-length (Float/parseFloat number) {:dots (count dots)}))
+        (throw (Exception. (format "Invalid note length: %s" length)))))
+
+    :else
+    (:value (note-length length))))
+
 (defattribute tempo
   "Current tempo. Used to calculate the duration of notes."
   :initial-val 120
-  :transform pos-num)
+  :transform (fn [val]
+               (cond
+                 (map? val)
+                 (fn [old]
+                   (* (:ratio val) old))
+
+                 :else
+                 (pos-num val))))
+
+(defn tempo
+  "Multi-arity function that allows for an additional parameter to represent
+   the tempo in terms of non-quarter notes (e.g.(tempo 2 60) <--> (tempo 120)).
+   Overwrites the single-argument function produced by defattribute"
+  ([val]
+  (set-attribute :tempo val))
+
+  ([note-length val]
+  (set-attribute :tempo (* (parse-note-length note-length) val))))
+
+(defn tempo!
+  "Global version of overwritten tempo function"
+  ([val]
+   (global-attribute :tempo val))
```

Comment:
(same deal here)

Reply:
In general, I'm not sure about including the music theory library, especially just for one function, at least for this pull request. Maybe you can open up another issue dealing with the integration.

Reply:
That's a good idea -- I'll do that 👍